services/worker-manager/src/providers/aws.js

```diff
@@ -0,0 +1,99 @@
+const {Provider} = require('./provider');
+const aws = require('aws-sdk');
+
+class AwsProvider extends Provider {
+  constructor({workerPool}) {
```

Comment:
Not sure I follow. I definitely see that `workerPool` gets passed in to the google provider's constructor

(and just to be clear: this code is not finished. This constructor should and will take in other arguments - I just don't want a whole bunch of unused variables in there at this point)

Reply:
The google provider gets `WorkerPool` which is the azure table of worker pools. I wonder if we should change our naming conventions to make this less confusing

Reply:
> the azure table of worker pools

do you mean it's the result of `WorkerPool.scan({})`? Should be `workerPools`, plural then. If it's the Entity, it should probably start with a capital?

Reply:
It isn't the result of `scan` it is `WorkerPool` itself. So that the provider can do things like `WorkerPool.load`. It is upper case in the constructor of the google provider.

Reply:
> It is upper case in the constructor of the google provider.

it is! I didn't notice that. I'll correct that, thanks